homeassistant/components/delijn/sensor.py

```diff
@@ -0,0 +1,107 @@
+"""Support for De Lijn (Flemish public transport) information."""
+import asyncio
+import logging
+
+import voluptuous as vol
+
+from homeassistant.components.sensor import PLATFORM_SCHEMA
+from homeassistant.exceptions import PlatformNotReady
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.entity import Entity
+
+_LOGGER = logging.getLogger(__name__)
+
+CONF_NEXT_PASSAGE = 'nextpassage'
+CONF_STOP_ID = 'stop_id'
+CONF_API_KEY = 'api_key'
+CONF_MAX_PASSAGES = 'max_passages'
+
+DEFAULT_NAME = 'De Lijn'
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_API_KEY): cv.string,
+    vol.Required(CONF_NEXT_PASSAGE): [{
+        vol.Required(CONF_STOP_ID): cv.string,
+        vol.Optional(CONF_MAX_PASSAGES, default=5): cv.positive_int}]
+})
+
+
+async def async_setup_platform(
+        hass, config, async_add_entities, discovery_info=None):
+    """Create the sensor."""
+    from pydelijn.api import Passages
+
+    api_key = config.get(CONF_API_KEY)
+    name = DEFAULT_NAME
+
+    session = async_get_clientsession(hass)
+
+    sensors = []
+    for nextpassage in config.get(CONF_NEXT_PASSAGE):
+        stop_id = nextpassage[CONF_STOP_ID]
+        max_passages = nextpassage[CONF_MAX_PASSAGES]
+        line = Passages(hass.loop, stop_id, max_passages, api_key, session)
+        sensors.append(DeLijnPublicTransportSensor(line, name))
+
+    tasks = [sensor.async_update() for sensor in sensors]
+    if tasks:
+        await asyncio.wait(tasks)
+    if not all(sensor._attributes for sensor in sensors):
```

Comment:
We shouldn't access private attributes outside the instance.

I suggest calling line.get_passages directly instead for all lines.

`async_add_entities(..., True)` will make sure `entity.async_update` is called during entity addition.

Reply:
Changes made in last commit